core/src/test/java/org/elasticsearch/script/expression/ExpressionScriptTests.java

```diff
@@ -385,8 +388,63 @@ public void testStringSpecialValueVariable() throws Exception {
             message = e.toString();
         }
         assertThat(message + "should have contained ExpressionScriptExecutionException",
-                   message.contains("ExpressionScriptExecutionException"), equalTo(true));
+                message.contains("ExpressionScriptExecutionException"), equalTo(true));
         assertThat(message + "should have contained text variable error",
-                   message.contains("text variable"), equalTo(true));
+                message.contains("text variable"), equalTo(true));
+    }
+
+    public void testExecutableScripts() throws Exception {
+        Map<String, Object> vars = new HashMap<>();
+        vars.put("a", 2.5);
+        vars.put("b", 3);
+        vars.put("xyz", -1);
+
+        Expression expr = JavascriptCompiler.compile("a+b+xyz");
+
+        ExpressionExecutableScript ees = new ExpressionExecutableScript(expr, vars);
+        assertEquals((Double) ees.run(), 4.5, 0.001);
+
+        ees.setNextVar("b", -2.5);
+        assertEquals((Double) ees.run(), -1, 0.001);
+
+        ees.setNextVar("a", -2.5);
+        ees.setNextVar("b", -2.5);
+        ees.setNextVar("xyz", -2.5);
+        assertEquals((Double) ees.run(), -7.5, 0.001);
+
+        String message = "";
+        try {
+            vars = new HashMap<>();
+            vars.put("a", 1);
+            ees = new ExpressionExecutableScript(expr, vars);
+            ees.run();
+        } catch (ScriptException se) {
+            message = se.getMessage();
+        }
+        assertThat(message + " should have contained number of variables", message.contains("number of variables"), equalTo(true));
```

Comment:
I think it would be cleaner to move the assert into the catch, and add a `fail("expected script exception")` after the call to `run()`.